modules/api/php/endpoints/candidate/visit/instrument/flags.class.inc

```diff
@@ -127,6 +126,104 @@ class Flags extends Endpoint implements \LORIS\Middleware\ETagCalculator
         );
     }
 
+    /**
+     * Handles a PUT request
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePUT(ServerRequestInterface $request) : ResponseInterface
+    {
+        // TODO :: Check permissions. How??
+        $this->_instrumentStatus = new \NDB_BVL_InstrumentStatus();
+        $this->_instrumentStatus->select($this->_instrument->commentID);
+
+        if (!$this->_instrument->determineDataEntryAllowed()) {
+            return new \LORIS\Http\Response\JSON\Forbidden(
+                'Can not update instruments that are flagged as complete.'
+            );
+        }
+
+        $data = json_decode((string) $request->getBody(), true);
+
+        if (!$this->_instrument->validate($data)) {
+            return new \LORIS\Http\Response\JSON\Forbidden(
+                'Could not update.'
+            );
+        }
+
+        try {
+            $this->_instrumentStatus->setDataEntryStatus(
+                $data['Flags']['Data_entry'] ?? null
+            );
+            $this->_instrumentStatus->setAdministrationStatus(
+                $data['Flags']['Administration'] ?? null
+            );
+            $this->_instrumentStatus->setValidityStatus(
+                $data['Flags']['Validity'] ?? null
+            );
+        } catch (\Throwable $e) {
+            error_log($e->getMessage());
+            return new \LORIS\Http\Response\JSON\InternalServerError();
+        }
+        return (new \LORIS\Http\Response())
+            ->withStatus(204);
+    }
+
+    /**
+     * Handles a PATCH request.
+     * Same as handlePatch but the instrument is not cleared before save
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    private function _handlePATCH(ServerRequestInterface $request): ResponseInterface
+    {
+        // TODO :: Check permissions. How??
+        $this->_instrumentStatus = new \NDB_BVL_InstrumentStatus();
+        $this->_instrumentStatus->select($this->_instrument->commentID);
+
+        if (!$this->_instrument->determineDataEntryAllowed()) {
```

Comment:
I don't think this is necessary
What needs to be done here is to mimic what is done at https://github.com/aces/Loris/blob/2457142f547d0e7beadae5c52749b51212dc77d0/php/libraries/NDB_BVL_InstrumentStatus_ControlPanel.class.inc#L152

The save function first calls _hasAccess, which might return false for some reason (you could make the reason  explicit here), then calls the setter(s). 

Note that the setters can throw exceptions that might not be InternalServerError (5XX) but Client error (4XX) instead.